<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <title>The source code</title>
  <link href="../resources/prettify/prettify.css" type="text/css" rel="stylesheet" />
  <script type="text/javascript" src="../resources/prettify/prettify.js"></script>
  <style type="text/css">
    .highlight { display: block; background-color: #ddd; }
  </style>
  <script type="text/javascript">
    function highlight() {
      document.getElementById(location.hash.replace(/#/, "")).className = "highlight";
    }
  </script>
</head>
<body onload="prettyPrint(); highlight();">
  <pre class="prettyprint lang-js"><span id='java-lang-ClassNotFoundException'>/**@class java.lang.ClassNotFoundException
</span>@extends java.lang.ReflectiveOperationException

 Thrown when an application tries to load in a class through its
 string name using:
 &lt;ul&gt;
 &lt;li&gt;The &lt;code&gt;forName&lt;/code&gt; method in class &lt;code&gt;Class&lt;/code&gt;.
 &lt;li&gt;The &lt;code&gt;findSystemClass&lt;/code&gt; method in class
     &lt;code&gt;ClassLoader&lt;/code&gt; .
 &lt;li&gt;The &lt;code&gt;loadClass&lt;/code&gt; method in class &lt;code&gt;ClassLoader&lt;/code&gt;.
 &lt;/ul&gt;
 &lt;p&gt;
 but no definition for the class with the specified name could be found.

 &lt;p&gt;As of release 1.4, this exception has been retrofitted to conform to
 the general purpose exception-chaining mechanism.  The &quot;optional exception
 that was raised while loading the class&quot; that may be provided at
 construction time and accessed via the {@link #getException}() method is
 now known as the &lt;i&gt;cause&lt;/i&gt;, and may be accessed via the {@link java.lang.Throwable#getCause()} method, as well as the aforementioned &quot;legacy method.&quot;

 @author  unascribed
 @see     java.lang.Class#forName(java.lang.String)
 @see     java.lang.ClassLoader#findSystemClass(java.lang.String)
 @see     java.lang.ClassLoader#loadClass(java.lang.String, boolean)
 @since   JDK1.0
*/
var ClassNotFoundException = {

<span id='java-lang-ClassNotFoundException-method-getException'>/**Returns the exception that was raised if an error occurred while
</span> attempting to load the class. Otherwise, returns &lt;tt&gt;null&lt;/tt&gt;.

 &lt;p&gt;This method predates the general-purpose exception chaining facility.
 The {@link java.lang.Throwable#getCause()} method is now the preferred means of
 obtaining this information.
@return {Object {java.lang.Throwable}} the &lt;code&gt;Exception&lt;/code&gt; that was raised while loading a class
@since 1.2
*/
getException : function(  ) {},

<span id='java-lang-ClassNotFoundException-method-getCause'>/**Returns the cause of this exception (the exception that was raised
</span> if an error occurred while attempting to load the class; otherwise
 &lt;tt&gt;null&lt;/tt&gt;).
@return {Object {java.lang.Throwable}} the cause of this exception.
@since 1.4
*/
getCause : function(  ) {},


};</pre>
</body>
</html>
